.github/workflows/unity.yml

```diff
@@ -0,0 +1,57 @@
+name: unity
+
+on:
+  push:
+    branches:
+      - master
+      - 'support/*'
+  pull_request: {}
+
+jobs:
+  unity:
+    name: Unity build
+
+    strategy:
+      fail-fast: false
+      matrix:
+        onoff:
+          - ON
+          - OFF
```

Comment:
I'd rather call the workflow "build" and use different sets of compile flags in the matrix.

This comment is part of my secret plan (well, not so much anymore now) regarding https://github.com/Icinga/icinga2/pull/9411#pullrequestreview-1054235644, where this job could then serve as a basis.

Reply:
_What_ for? For a use case which we don’t even agree on yet?

Reply:
Well it's not only about that. For now, the name "unity" describes the purpose of the job right now. Implementing my change would make it more flexible without drawbacks on the current use. You could also easily add more combinations of flags, like (non-)debug, (non-)systemd. Sure, not all combinations and not necessarily right now, but why not make the workflow easily extendable?

Reply:
Do you _want_ me to test (non-)debug, (non-)systemd?

Reply:
I wouldn't overload this check for now, we're putting enough stress on the CI already with our checks already.

Well probably do some overhaul of the package jobs in the not so distant future, so let's see where this brings us and what extra jobs we might want then.

Reply:
Even if I don't want you to add more variations of compile flags right now, I still think it's a good idea to start with a more generic workflow name like "build".

Reply:
What about GCC version + Boost version as one additional parameter, one per supported OS? Then we can throw away the .deb, .rpm, etc. GHAs, test the SLES/RHEL/... GCC version + Boost version even on foreign PRs and save build time by disabling LTO.

Reply:
Sounds like a good idea. Do you already have an idea how you'd get these versions, i.e. would you try to install them on the Ubuntu system (already has [4 GCC versions available](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#language-and-runtime) but we'd probably want to test with something older that that and also no Boost apart from the Ubuntu repos) or do it inside Docker (would this mean maintaining another set of custom images)?

Reply:
https://hub.docker.com/_/gcc/tags?page=1&ordering=-last_updated
:)

Reply:
And what about Boost?

Reply:
From source.

Reply:
Forget it, there's also libstdc++, etc.

Reply:
Do you still insist on this? (Other blockers done, no add. matrix items in sight.)

Reply:
Insist on which part? I still think something like "build" would be a better name for the job.

Reply:
Do you _insist_ on this? If yes, do you have suggestions for additional matrix vars (given we're not gonna replace our package build jobs yet)?